ruby_programming/advanced_ruby/blocks.md

```diff
@@ -287,30 +295,40 @@ a_lambda.call
 # => 1
 ~~~
 
-A Proc object returns from the context in which it is called. If you are in the top level context (outside of a class and method) then you'll get an error because you can't return out of the very top level context, as there is no caller to return to.
+In non-lambda procs, `return` means exit from embracing method in which it was *defined* and will throw `LocalJumpError` if invoked outside the method. Check the [documentation](https://ruby-doc.org/core-2.7.0/Proc.html#class-Proc-label-Lambda+and+non-lambda+semantics) about it. In the top level context (not inside any method) a `return` will exit the program.
+
+Calling the proc (with an explicit `return`) will return from the method it was defined in:
 
 ~~~ruby
-a_proc = Proc.new { return }
+def execute_proc(proc_inside_method)
+  puts proc_inside_method.call # will produce no output!
+end
 
-a_proc.call
+def define_proc
+  proc_inside_method = proc { return 'explicit return' }
+  puts 'Calling the proc inside "execute_proc" will return from "define_proc".'
```

Comment:
I get the point you are trying to make, but this isn't explained well enough for someone new to blocks.